extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/KryoFeatureDataAdapter.java

```diff
@@ -128,13 +131,32 @@ public AdapterPersistenceEncoding encode(
 		final byte[] typeNameBytes = StringUtils.stringToBinary(typeName);
 		final byte[] visibilityManagementClassNameBytes = StringUtils.stringToBinary(fieldVisibilityManagement.getClass().getCanonicalName());
 		final byte[] encodedTypeBytes = StringUtils.stringToBinary(encodedType);
-		final ByteBuffer buf = ByteBuffer.allocate(encodedTypeBytes.length + typeNameBytes.length + visibilityManagementClassNameBytes.length + adapterId.getBytes().length + 16);
+		final ByteBuffer buf = ByteBuffer.allocate(encodedTypeBytes.length + typeNameBytes.length + visibilityManagementClassNameBytes.length + adapterId.getBytes().length + 20);
+		byte[] attrBytes = new byte[0];
+
+		final SimpleFeatureUserDataConfigurationSet userDataConfiguration = new SimpleFeatureUserDataConfigurationSet();
+		userDataConfiguration.addConfigurations(new TimeDescriptorConfiguration(
+				featureType));
+		userDataConfiguration.addConfigurations(new SimpleFeatureStatsConfigurationCollection(
+				featureType));
+		try {
+			attrBytes = StringUtils.stringToBinary(userDataConfiguration.asJsonString());
+		}
+		catch (final IOException e) {
+			LOGGER.error(
+					"Failure to encode simple feature user data configuration",
+					e);
+		}
+
+		buf.putInt(0); // a signal for the new version
```

Comment:
Using the int since I can 'reuse' its value if it turns out to be the name length.